src/streaming/utils/Capabilities.js

```diff
@@ -193,6 +196,31 @@ function Capabilities() {
     function supportsEssentialProperty(ep) {
         let supportedEssentialProps = settings.get().streaming.capabilities.supportedEssentialProperties;
 
+        // we already took care of these descriptors with the codecs check
+        // let's bypass them here
+        if (settings.get().streaming.capabilities.useMediaCapabilitiesApi && settings.get().streaming.capabilities.filterVideoColorimetryEssentialProperties) {
```

Comment:
Do we need to check for `useMediaCapabilitiesApi` and `filterVideoColorimetryEssentialProperties` in this case? We already filtered the AdaptationSets and Representations as part of the codec check if these two settings are enabled. Shouldn't this check be purely based on whether `filterUnsupportedEssentialProperties` is enabled?

Reply:
Yes, we need this here:

`filterUnsupportedEssentialProperties` is evaluated to level higher in call stack. (CapabilitiesFilter.js:57)
Only if that is enabled, this code you commented on here gets executed.

When both Settings-conditions (`useMediaCapabilitiesApi` and `filterVideoColorimetryEssentialProperties`) are enabled, the if-statement registers HDR-Properties as "known"/"supported".
Otherwise the parent `AdaptationSet`s/`Representation`s would get removed since line 225 would evaluate to `false`.

Another option would be to remove the `EssentialProperty` elements from the manifest-object after they were converted to MediaCapabilitiesAPI calls; IMHO not the desired approach.

Reply:
Thanks, I think I understand the idea here now: As we did the check with the Media Capabilities API previously, we are now allowing any values instead of the limited set of values provided via `Settings.js`.

If we did not check with the Media Capabilities API before, then we only check against the schemeId/value combinations allowed via `Settings.js`.

One more request: Can we move the filtering logic to a small helper function to avoid duplicating code. It looks like only the input (`EssentialHDRProperties`) changes but the following is duplicated:

````js
supportedEssentialProps = supportedEssentialProps.filter(p => {
    return !(p.schemeIdUri && (EssentialHDRProperties.some(ep=>ep.schemeIdUri===p.schemeIdUri)));
});
supportedEssentialProps.push(...EssentialHDRProperties);
````